src/BlazingPizza.Client/ServerAuthenticationStateProvider.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Net.Http;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+
+namespace BlazingPizza.Client
+{
+    public class ServerAuthenticationStateProvider : AuthenticationStateProvider
+    {
+        private readonly HttpClient http;
+
+        public ServerAuthenticationStateProvider(HttpClient httpClient)
+        {
+            http = httpClient;
+        }
+
+        public override async Task<AuthenticationState> GetAuthenticationStateAsync()
+        {
+            var userInfo = await http.GetJsonAsync<UserInfo>("user");
+
+            var identity = userInfo.IsAuthenticated
+                ? new ClaimsIdentity(new[] { new Claim(ClaimTypes.Name, userInfo.Name) }, "serverauth")
```

Comment:
I realize it doesn't matter but it bothers me a little seeing fake auth types here. Does it make sense to serialize this from the server?

Reply:
As general guidance I’d advise people to let the client only know the minimum subset of authentication data that it needs. For example, don’t serialise our *all* the claims. Since the client doesn’t need to know what auth type name the server uses, why tell it? It’s just unfortunate our API requires an auth type name to be specified.